c++/src/MemoryPool.cc

```diff
@@ -134,9 +130,6 @@ namespace orc {
   template <>
   void DataBuffer<char>::resize(uint64_t newSize) {
     reserve(newSize);
-    if (newSize > currentSize_) {
```

Comment:
Lines below seem to be unrelated changes. Could you please revert them? I remember there was an effort to remove these but caused issues to the downstream.

Reply:
Do you think it makes sence if we add a build option like `MEMSET_AFTER_RESIZE` with false as default value

Reply:
> Do you think it makes sence if we add a build option like `MEMSET_AFTER_RESIZE` with false as default value.

already reverted. But I still think it is worth to try. If you think the solution is ok, I'll do it in another newe pr.